### PR TITLE
tests: Add InvalidImageLayout tests

### PR DESCRIPTION
**Pull request by tobine**
Addresses #432
These hit 16/18 cases for INVALID_IMAGE_LAYOUT enum in core_validation.
The other 2 cases were already hit by object_tracker and another test.

### CODE REVIEW

**Comment by mark-lunarg** [comment]
+1

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -6142,6 +6142,177 @@ TEST_F(VkLayerTest, VtxBufferBadIndex) {
...
+    rpci.subpassCount = 1;
+    rpci.pSubpasses = &subpass;
+    rpci.attachmentCount = 1;
+    VkAttachmentDescription attach_desc = {};
+    attach_desc.format = VK_FORMAT_UNDEFINED;
+    rpci.pAttachments = &attach_desc;
+    VkRenderPass rp;
+    m_errorMonitor->SetDesiredFailureMsg(
+        VK_DEBUG_REPORT_PERFORMANCE_WARNING_BIT_EXT,
+        "Layout for input attachment is GENERAL but should be READ_ONLY_OPTIMAL.");
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: I'm not sure about complaining about attachments in GENERAL layout. The current code isn't smart enough to /not/ emit the perf warning in cases where GENERAL is the only valid layout -- attachment used as both input attachment and color attachment, for example.

Comment (tobine) [inline tests/layer_validation_tests.cpp]: This set of tests is about hitting the cases that are in place.
You bring up a good point about perf warning in GENERAL case, but I believe it's separate from this test commit.
I'll file a separate ? issue where we can discuss what to do with these GENERAL layout perf warnings.

**Comment by tobine** [comment]
Filed #508 to address @chrisforbes point regarding GENERAL layout perf warning.

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**